CBLClient/Apps/CBLTestServer-iOS/CBLTestServer-iOS/Server/PredictiveQueriesRequestHandler.swift

```diff
@@ -0,0 +1,141 @@
+            do{
+                try queryResult.execute();
+            }
+            catch {
+              return error.localizedDescription
+            }
+            
+        case "predictiveQuery_getNumberOfCalls":
+            let model: EchoModel = args.get(name: "model")!
+            return model.numberOfCalls
+            
+        case "predictiveQuery_euclideanDistance":
+            let database: Database = args.get(name: "database")!
+            let key1: String = args.get(name: "key1")!
+            let key2: String = args.get(name: "key2")!
+            let key3_distance = Function.euclideanDistance(between: Expression.property(key1),
+                                                      and: Expression.property(key2))
+            let queryResult = QueryBuilder
+                .select(SelectResult.expression(key3_distance))
+                .from(DataSource.database(database))
+            var resultArray = [Any]()
+            
+            for row in try queryResult.execute() {
+                resultArray.append(row.toDictionary())
+            }
+            
+            return resultArray
+            
+        case "predictiveQuery_squaredEuclideanDistance":
+            let database: Database = args.get(name: "database")!
+            let key1: String = args.get(name: "key1")!
+            let key2: String = args.get(name: "key2")!
+            let distance = Function.squaredEuclideanDistance(between: Expression.property(key1),
+                                                           and: Expression.property(key2))
+            let queryResult = QueryBuilder
+                .select(SelectResult.expression(distance))
+                .from(DataSource.database(database))
+            var resultArray = [Any]()
+            
+            for row in try queryResult.execute() {
+                resultArray.append(row.toDictionary())
+            }
+            
+            return resultArray
+            
+        case "predictiveQuery_cosineDistance":
+            let database: Database = args.get(name: "database")!
+            let key1: String = args.get(name: "key1")!
+            let key2: String = args.get(name: "key2")!
+            let distance = Function.cosineDistance(between: Expression.property(key1),
+                                                             and: Expression.property(key2))
+            let queryResult = QueryBuilder
+                .select(SelectResult.expression(distance))
+                .from(DataSource.database(database))
+            var resultArray = [Any]()
+            
+            for row in try queryResult.execute() {
+                resultArray.append(row.toDictionary())
+            }
+            
+            return resultArray
+        
+        default:
+            throw RequestHandlerError.MethodNotFound(method)
+        }
+        return PredictiveQueriesRequestHandler.VOID
+    }
+}
+
+class EchoModel: PredictiveModel {
```

Comment:
is get method for name intentionally skipped?

Reply:
added